File: 27.remove-element.py
#
# @lc app=leetcode id=27 lang=python
#
# [27] Remove Element
#

# @lc code=start
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        elems = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[elems] = nums[i]
                elems += 1

        return elems

        
# @lc code=end

File: solutions_java/arrays/27.remove-element.java
/*
 * @lc app=leetcode id=27 lang=java
 *
 * [27] Remove Element
 */

// @lc code=start
class Solution {
    public int removeElement(int[] nums, int val) {
        int start = 0;
        int end = nums.length - 1;

        while (start <= end) {
            if (nums[start] == val) {
                while (start<end && nums[end] == val) {
                    end--;
                }
                nums[start] = nums[end];
                end--;
            }
            start++;
        }
        return end+1;
    }

}
// @lc code=end

